### ISSUE TITLE
Resuming with specified weights (last.pt) doesn't work properly.

### ISSUE THREAD

**Issue by alexk-ede** [ultralytics/yolov5#9525]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and found no similar bug report.

### YOLOv5 Component

Training

### Bug

Resuming with specified weights (last.pt) doesn't work.
I have several different trainings that I did pause and have now a few runs_yolo5m, runs_yolo5n, etc folders ... 
and despite starting with --resume --weights 'runs_yolo5m/train/exp/weights/last.pt' it always goes to the runs_yolo5n folder (which was the last one running and that got stopped)

### Environment

(env doesn't really matter, but anyway)
- yolo: latest checkout
- os ubuntu 22.04
- python 3.10.4 

### Minimal Reproducible Example

I'm quite sure the problem is in 
https://github.com/ultralytics/yolov5/blob/77dcf55168d59131f75b8187c6be27172eec00ec/train.py#L487

opt.resume is always a bool. So no way this will be ever true.
I changed both to opt.weights instead, then it makes more sense with checking for str.
` last = Path(check_file(opt.weights) if isinstance(opt.weights, str) else get_latest_run()) `
and then it resumes properly for me.

Can somebody confirm ?

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
👋 Hello! Thanks for asking about **resuming training**. YOLOv5 🚀 Learning Rate (LR) [schedulers](https://github.com/ultralytics/yolov5/blob/aa542ce6a65658ff931fee9bbab77c0145c152f0/train.py#L175-L181) follow predefined LR curves for the fixed number of `--epochs` defined at training start (`default=300`), and are designed to fall to a minimum LR on the final epoch for best training results. For this reason you can not modify the number of epochs once training has started.

<img width="850" alt="Screenshot 2022-04-10 at 11 11 51" src="https://user-images.githubusercontent.com/26833433/162611202-4e0b611f-6cca-4e43-94fc-bbc44e6447f8.png">

If your training was **interrupted** for any reason you may continue where you left off using the `--resume` argument. If your training fully completed, you can start a **new** training from any model using the `--weights` argument. Examples:

## Resume Single-GPU

You may not change settings when resuming, and **no additional arguments** other than `--resume` should be passed, with an optional path to the checkpoint you'd like to resume from. If no checkpoint is passed the most recently updated `last.pt` in your `yolov5/` directory is automatically found and used:
```bash
python train.py --resume  # automatically find latest checkpoint (searches yolov5/ directory)
python train.py --resume path/to/last.pt  # specify resume checkpoint
```

## Resume Multi-GPU

[Multi-GPU DDP trainings](https://docs.ultralytics.com/yolov5/tutorials/multi_gpu_training) must be resumed with the same GPUs and DDP command, i.e. assuming 8 GPUs:
```bash
python -m torch.distributed.run --nproc_per_node 8 train.py --resume  # resume latest checkpoint
python -m torch.distributed.run --nproc_per_node 8 train.py --resume path/to/last.pt  # specify resume checkpoint
```

## Start from Pretrained

If you would like to **start training** from a fully trained model, use the `--weights` argument, not the `--resume` argument:
```bash
python train.py --weights path/to/best.pt  # start from pretrained model
```

Good luck 🍀 and let us know if you have any other questions!

**Comment by alexk-ede** [comment]
Ahh I see, ok, seems I mixed it up with the action='store_true' from the other lines and that's why I thought it can only be bool.
If pass the weights pt via the resume argument it works indeed. 
Thanks.

**Comment by glenn-jocher** [comment]
@alexk-ede you're welcome! I'm glad we could clear that up. If you have any more questions or need further assistance, feel free to ask. Good luck with your training!
